### PR TITLE
Building a Calculator: A JavaScript project for beginners

### PR DESCRIPTION
**Pull request by Nomzy-kush**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
This fixes #2027

**Comment by adrianmurage** [comment]
@WanjaMIKE 
I see you have 8 PRs assigned to you
Could you let me take this one?

**Comment by WanjaMIKE** [comment]
@adrianmurage No, I have two PRs. This one is already in progress. Thanks

File: articles/building-a-calculator-a-javascript-project-for-beginners/index.md [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+  <head>
+    <meta charset="utf-8">
+    <title>Calculator</title>
+    <link href="Calculator with Js\style.css" rel="stylesheet">
+    <script src="Calculator with JS\script.js" defer></script>
+```
+
+The first thing you need to do is add all the different elements using “grid” to lay everything out nice and neat. You will then need to create a class naming it "calculator-grid". You will put all the different elements, the screen, and all of the other buttons inside the application. Below is the HTML code for the calculator.
```
Comment (WanjaMIKE) [inline articles/building-a-calculator-a-javascript-project-for-beginners/index.md:30]: You already said the `first thing ... ` before. Replace it with another term

File: articles/building-a-calculator-a-javascript-project-for-beginners/index.md [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+ You can now hook that variable up by implementing the Delete button. Below is the code to do that:
+```javascript
+deleteButton.addEventListener('click', button => {
+  calculator.delete()
+  calculator.updateDisplay()
+})
+```
+If you type a long string of characters, you can delete them one by one using the Delete button. The calculator is now fully functional, but the display still needs some work.
+
+#####updateDisplay Function
```
Comment (WanjaMIKE) [inline articles/building-a-calculator-a-javascript-project-for-beginners/index.md:390]: Only use level 3 headings ```###```

File: articles/building-a-calculator-a-javascript-project-for-beginners/index.md [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+```javascript
+updateDisplay() {
+    if (this.operation != null) {
+      this.previousOperandTextElement.innerText =
+        `${this.getDisplayNumber(this.previousOperand)} ${this.operation}`
+```
+On your calculator, you may notice that when you type a string of numbers, there are no commas to really make the numbers more definitive. Well, you can change that by using a helper function and calling it `getDisplayNumber(number)`. Then you will return that number but in the form of a itssplay value. You are going to call a function in the updateDisplay() say "this.GetDisplayNumber" then pass it in the current operand. You’ll also do the exact same thing but you'll do it for "this.previousOperand" and now the changes you make in this function, are going to be reflected in both the previous and the current operand values. Here's what the entire updateDisplay() function code would look like:
```
Comment (WanjaMIKE) [inline articles/building-a-calculator-a-javascript-project-for-beginners/index.md:398]: Try to use shorter sentences. They are usually more understandable

**Comment by WanjaMIKE** [comment]
@Nomzy-kush make sure you format your article correctly. add a space after ```###```. Only use h3 headings in your article.

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Please check the above comments

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by WanjaMIKE — APPROVED** [review]
A very detailed tutorial. Thumps Up. Take note of the changes I have made across the article. Thanks

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Pull request closed without merging**